symengine/rational.h

```diff
@@ -20,8 +20,17 @@ class Rational : public Number
 
 public:
     IMPLEMENT_TYPEID(RATIONAL)
-    //! Constructor of Rational class
+//! Constructor of Rational class
+#if SYMENGINE_INTEGER_CLASS == SYMENGINE_BOOSTMP
```

Comment:
Yes, we should have all 4 methods. This was probably there because we couldn't figure out how to enable move semantics of `mpz_class` without enabling other C++11 features. (SymEngine supported gcc 4.6 back then).  `i{std::move(i)}` is a newer addition, I think.
Anyways, we should have all 4 methods you mentioned.